modules/core/src/main/java/org/apache/ignite/internal/processors/platform/client/ClientRequestHandler.java

```diff
@@ -115,7 +116,14 @@
         int status = e instanceof IgniteClientException ?
             ((IgniteClientException)e).statusCode() : ClientStatus.FAILED;
 
-        return new ClientResponse(req.requestId(), status, e.getMessage());
+        boolean fullStack = false;
+
+        if (X.getCause(e) != null || !X.getSuppressedList(e).isEmpty()) {
```

Comment:
This is a suspicion condition, may we shall print stack always.

Reply:
In ideal - i like idea to show only one error message and probably error code, thus i prefer to store this logic where it possible.

Reply:
What does the condition (X.getCause(e) != null || !X.getSuppressedList(e).isEmpty()) mean?